doc/source/serve/performance.md

```diff
@@ -159,3 +161,10 @@ hitting the same queuing issue mentioned above, you might want to increase
 `max_concurrent_queries`. Serve sets a low number (100) by default so the client gets
 proper backpressure. You can increase the value in the deployment decorator; e.g.
 `@serve.deployment(max_concurrent_queries=1000)`.
+
+(serve-performance-e2e-timeout)=
+### Set an end-to-end request timeout
+
+By default, Serve lets client HTTP requests run to completion no matter how long they take. Slow requests could bottleneck the replica processing them while other requests wait in a queue. It's recommended that you set an end-to-end timeout, so slow requests can be terminated and retried at another replica.
```

Comment:
Not really related to this PR, but if this is our recommendation, I wonder if we should make it the default, since many users might not read this part of the doc.

Reply:
The issue with making it the default is it's technically a breaking change for users that currently have intentionally long-running requests :(